types/react-native/index.d.ts

```diff
@@ -8970,11 +8970,13 @@ export interface ComponentInterface<P> {
  * Common types are lined up with the appropriate prop differs with
  * `TypeToDifferMap`.  Non-scalar types not in the map default to `deepDiffer`.
  */
-export function requireNativeComponent<P>(
+export function requireNativeComponent<P, NP = {}>(
     viewName: string,
     componentInterface?: ComponentInterface<P>,
-    extraConfig?: { nativeOnly?: any }
-): React.ComponentClass<PropTypes.InferProps<PropTypes.ValidationMap<P>>>;
+    extraConfig?: { nativeOnly?: NP }
+): React.ComponentClass<
+    Partial<PropTypes.InferProps<PropTypes.ValidationMap<P>>> & { [K in keyof NP]?: any}
```

Comment:
What is the last mapped typed addition for?

Reply:
The "nativeOnly" map, which looks like `{ prop1: true, prop2: true, ...}`, specifies props that are defined by the native Obj-C/Java module but aren't declared in the JS component's `propTypes`. (Also, all these prop type declarations are going away in RN anyway so expect to see a PR that deletes my recent PRs :P)